docs(self-managed): align storage performance recommendations across providers (#9085)
camunda/camunda-docs: docs/self-managed/reference-architecture/kubernetes.md

#### Minimum cluster requirements

> The following are suggested minimum requirements to get started. There is no one-size-fits-all configuration: sizing depends heavily on your specific use cases and workload, so treat these values as a baseline rather than a strict requirement. Refer to [sizing your environment](/components/best-practices/architecture/sizing-your-environment.md) and [Zeebe resource planning](/self-managed/components/orchestration-cluster/zeebe/operations/resource-planning.md), and conduct benchmarking to determine your exact needs.

Reviewer: Wording drift between versions: `docs/` says "treat these values as a baseline", while the 8.9 file says "a starting baseline". These two files should be identical — pick one.

Minor: this sentence and the `:::note` two lines below both state "baseline, not a strict requirement". Consider trimming one to avoid repeating the same caveat.

Reviewer 2: Apologies, the wording drift between the two intros was introduced by me. I removed starting because it's redundant.

> The following are suggested minimum requirements to get started. There is no one-size-fits-all configuration: sizing depends heavily on your specific use cases and workload, so treat these values as a starting baseline rather than a strict requirement. Refer to [sizing your environment](/components/best-practices/architecture/sizing-your-environment.md) and [Zeebe resource planning](/self-managed/components/orchestration-cluster/zeebe/operations/resource-planning.md), and conduct benchmarking to determine your exact needs.

Reviewer 2: ```suggestion
The following are suggested minimum requirements to get started. There is no one-size-fits-all configuration: sizing depends heavily on your specific use cases and workload, so treat these values as a baseline rather than a strict requirement. Refer to [sizing your environment](/components/best-practices/architecture/sizing-your-environment.md) and [Zeebe resource planning](/self-managed/components/orchestration-cluster/zeebe/operations/resource-planning.md), and conduct benchmarking to determine your exact needs.
```

- **4 Kubernetes nodes**
  - CPU: 4 modern cores
  - Memory: 16 GiB
- **Persistent volumes**

>   - 3,000 IOPS

Author: ```suggestion
  - 3,000 IOPS baseline
```

>   - 125 MiB/s throughput

Reviewer: Consistency: this reads "125 MiB/s throughput", but the line just above now says "3,000 IOPS **baseline**" and every platform section (OpenShift, EKS, AKS, GKE) says "125 MiB/s throughput **baseline**". Suggest making this "125 MiB/s throughput baseline" too.

Same line in the 8.9 file — change both together so the versions stay in sync (a one-click suggestion would only touch this file and re-introduce a divergence, so I'm leaving it as a note).

Reviewer: Applied in `c05813f8f`. Thanks!

>   - 125 MiB/s throughput

Reviewer: Consistency with the platform sections, which all say "baseline":

```suggestion
  - 125 MiB/s throughput baseline
```

Reviewer: Applied in `c05813f8f`. Thanks!

  - 32 GiB
  - _Avoid burstable disk types_

> :::note Storage performance figures are a baseline, not a strict requirement

Reviewer: The note says these targets "apply across all providers (… GKE …)", but the GKE section below never states 3,000 IOPS — it says IOPS "varies based on volume size" and only references "> 1,000 IOPS". Please either add a `3,000 IOPS baseline` line to GKE (mirroring EKS/AKS) or soften this sentence so it doesn't promise a literal uniform 3,000 IOPS across every provider.

Grammar nit: "…and generic Kubernetes), there is no provider-specific benchmark behind them." is a comma splice — use a semicolon or a full stop.

(Same note in the 8.9 file, line 193.)

> The storage performance figures in this section and in the platform-specific sections below (for example, 3,000 IOPS and 125 MiB/s throughput) are a starting point, not hard requirements validated by benchmarking. The same targets apply across all providers (OpenShift, EKS, AKS, GKE, and generic Kubernetes), there is no provider-specific benchmark behind them. Actual needs vary with your workload, throughput, data retention, and exporter load. For production sizing, see [Self-Managed resource planning](/components/best-practices/architecture/sizing-self-managed.md) and benchmark against your own workload.

Reviewer: Comma splice — "…(OpenShift, EKS, AKS, GKE, and generic Kubernetes), there is no provider-specific benchmark behind them." joins two independent clauses with a comma. Use a semicolon or a full stop, for example: "…and generic Kubernetes); there is no provider-specific benchmark behind them."

The same sentence exists in the 8.9 file — please fix both in the same commit so the versions stay identical.

Reviewer: Applied in `c05813f8f`. Thanks!

> The storage performance figures in this section and in the platform-specific sections below (for example, 3,000 IOPS and 125 MiB/s throughput) are a starting point, not hard requirements validated by benchmarking. The same targets apply across all providers (OpenShift, EKS, AKS, GKE, and generic Kubernetes), there is no provider-specific benchmark behind them. Actual needs vary with your workload, throughput, data retention, and exporter load. For production sizing, see [Self-Managed resource planning](/components/best-practices/architecture/sizing-self-managed.md) and benchmark against your own workload.

Reviewer: Comma splice — two independent clauses joined by a comma. A semicolon fixes it:

```suggestion
The storage performance figures in this section and in the platform-specific sections below (for example, 3,000 IOPS and 125 MiB/s throughput) are a starting point, not hard requirements validated by benchmarking. The same targets apply across all providers (OpenShift, EKS, AKS, GKE, and generic Kubernetes); there is no provider-specific benchmark behind them. Actual needs vary with your workload, throughput, data retention, and exporter load. For production sizing, see [Self-Managed resource planning](/components/best-practices/architecture/sizing-self-managed.md) and benchmark against your own workload.
```

Reviewer: Applied in `c05813f8f`. Thanks!

> The storage performance figures above and in the distribution- and cloud-specific sections below (for example, 3,000 IOPS and 125 MiB/s throughput) are a baseline to get started, not authoritative benchmarked requirements. There is no dedicated, provider-specific benchmark behind them, and the same target applies regardless of provider (OpenShift, EKS, AKS, GKE, or generic Kubernetes). Actual needs vary with your workload, throughput, data retention, and exporter load. For production sizing, see [Self-Managed resource planning](/components/best-practices/architecture/sizing-self-managed.md) and benchmark against your own workload.

Reviewer 2: ```suggestion
The storage performance figures in this section and in the platform-specific sections below (for example, 3,000 IOPS and 125 MiB/s throughput) are a starting point, not hard requirements validated by benchmarking. The same targets apply across all providers (OpenShift, EKS, AKS, GKE, and generic Kubernetes), there is no provider-specific benchmark behind them. Actual needs vary with your workload, throughput, data retention, and exporter load. For production sizing, see [Self-Managed resource planning](/components/best-practices/architecture/sizing-self-managed.md) and benchmark against your own workload.
```

#### Minimum cluster requirements

The following are suggested minimum requirements to get started. There is no one-size-fits-all configuration: sizing depends heavily on your specific use cases and workload, so treat these values as a baseline rather than a strict requirement. Refer to [sizing your environment](/components/best-practices/architecture/sizing-your-environment.md) and [Zeebe resource planning](/self-managed/components/orchestration-cluster/zeebe/operations/resource-planning.md), and conduct benchmarking to determine your exact needs.

- **4 Kubernetes nodes**
  - CPU: 4 modern cores
  - Memory: 16 GiB
- **Persistent volumes**
  - 3,000 IOPS baseline
  - 125 MiB/s throughput baseline

>   - 32 GiB

Reviewer: Fixed in 86bf318fd — `32 GiB` is now labeled "minimum capacity", and the section note explains that on providers where performance scales with disk size (for example, GKE `pd-ssd`), reaching the IOPS/throughput baseline can require a larger disk.

  - SSD-backed volumes only. HDD-backed volumes are not supported.

>   - Avoid burstable volume types unless they can sustain the required IOPS continuously without relying on burst credits.

Reviewer: Fixed in c9576f331 — reworded "required IOPS" to "target IOPS" to match the "baseline, not a strict requirement" framing.

:::note Storage performance figures are a baseline, not a strict requirement

> The storage performance figures in this section and in the platform-specific sections below (for example, 3,000 IOPS and 125 MiB/s throughput) are a starting point, not hard requirements validated by benchmarking. The same targets apply across all providers (OpenShift, EKS, AKS, GKE, and generic Kubernetes); there is no provider-specific benchmark behind them. Actual needs vary with your workload, throughput, data retention, and exporter load. For production sizing, see [Self-Managed resource planning](/components/best-practices/architecture/sizing-self-managed.md) and benchmark against your own workload.

Reviewer: Updated the PR description to state the change covers `/docs` and the `8.9`, `8.8`, and `8.7` versioned docs. The 8.8/8.7 backports are intentional, to keep the storage baseline consistent with #9078 across all maintained versions.

- Instance type: 4 vCPUs (x86_64, >3.1 GHz), 16 GiB memory
- Number of dedicated nodes: 4
- Volume type: SSD

>   - 3,000 IOPS baseline

Reviewer: Good change — removing the unsupported `1,000 MB/s` and aligning to `3,000 IOPS` / `125 MiB/s` is correct.

Scope check: the same `1,000 MB/s` figure is still published in 8.8 (`kubernetes.md` L268) and 8.7 (L220), which this PR doesn't touch. If those versions are still maintained, the disowned number stays live there — backport, or confirm 8.9+ is the intended scope? (#9078 chose to cover all four versions.)

(Same change in the 8.9 file, line 282.)

Reviewer: Done in `77415bce8` — extended the storage baseline to **8.8 and 8.7** so all maintained versions stay aligned (previously next/8.9 only):

- Generic Persistent volumes: `1,000 IOPS` → `3,000 IOPS baseline` + `125 MiB/s throughput baseline`, plus the same "baseline, not a strict requirement" `:::note`.
- OpenShift / EKS / AKS / GKE aligned to `3,000 IOPS` / `125 MiB/s` baseline.
- 8.7 keeps its existing wording/links, and its note points to [sizing your environment](/components/best-practices/architecture/sizing-your-environment.md) because `sizing-self-managed.md` does not exist in 8.7.

This also matches the version coverage of #9078 (next/8.9/8.8/8.7). Leaving this thread open for you to confirm you're happy with the scope expansion on your PR — feel free to resolve once reviewed.

Reviewer: Confirmed — the scope expansion to 8.8/8.7 is intended, and the branch is now merged with main (the storage baseline and the #9078 SSD/HDD storage-type rules are combined). Resolving.

- Number of Kubernetes nodes: 4
- Volume type: Premium SSD v2
  - 3,000 IOPS baseline
  - 125 MiB/s throughput baseline
  - Several [known limitations](https://learn.microsoft.com/en-us/azure/virtual-machines/disks-types#premium-ssd-v2-limitations), e.g., lack of [Azure Backup support](https://learn.microsoft.com/en-us/azure/backup/disk-backup-support-matrix#limitations)
- Volume alternative: Premium SSD
  - IOPS performance [varies based on volume size](https://learn.microsoft.com/en-us/azure/virtual-machines/disks-types#premium-ssds)
  - Minimum 256 GiB (P15) for > 1,000 IOPS
- Unsupported volume types: Standard HDD is not supported.

> - Volume alternative caveat: Standard SSD is supported if sized to sustain the target IOPS and throughput continuously without relying on bursting.

Reviewer: Fixed in 86bf318fd — replaced "Minimum 256 GiB (P15) for > 1,000 IOPS" with "size the disk to sustain the target IOPS and throughput baseline", aligning the Premium SSD alternative with the 3,000 IOPS / 125 MiB/s baseline. Applied the same treatment to the EKS gp2 alternative.

- Instance type: n(1|2)-standard-4 (4 vCPUs, 15 / 16 GiB memory)
- Number of Kubernetes nodes: 4
- Volume type: Performance (SSD) persistent disks
  - 3,000 IOPS baseline
  - 125 MiB/s throughput baseline

>   - Minimum 34 GiB for > 1,000 IOPS

Reviewer: Addressed in fecbab9de — aligned the GKE sizing line to the stated baseline: `Minimum 100 GiB for 3,000 IOPS` (GCP pd-ssd delivers ~30 IOPS/GiB, so 100 GiB ≈ 3,000 IOPS). Applied across next/8.9/8.8/8.7.

- Number of Kubernetes nodes: 4
- Volume type: Performance (SSD) persistent disks

>   - IOPS performance [varies based on volume size](https://cloud.google.com/compute/docs/disks/performance#performance_factors)

Author: ```suggestion
  - 3,000 IOPS baseline
```

>   - 125 MiB/s throughput baseline

Reviewer: GKE now lists "125 MiB/s throughput baseline" but no matching IOPS baseline — its only IOPS reference is "Minimum 34 GiB for > 1,000 IOPS", which contradicts the new note's "3,000 IOPS … across all providers". Adding `- 3,000 IOPS baseline` here (as in the EKS and AKS sections) would make GKE consistent with the note.

(Same in the 8.9 file, line 382.)

- Instance type: n(1|2)-standard-4 (4 vCPUs, 15 / 16 GiB memory)
- Number of Kubernetes nodes: 4
- Volume type: Performance (SSD) persistent disks
  - 3,000 IOPS baseline
  - 125 MiB/s throughput baseline

>   - Minimum 100 GiB for 3,000 IOPS

Reviewer: Fixed in c9576f331 — replaced with `~260 GiB to sustain the 3,000 IOPS and 125 MiB/s baseline`. On pd-ssd throughput scales at ~0.48 MB/s per GiB, so throughput is the binding constraint (125 MiB/s ≈ 260 GiB > 100 GiB for 3,000 IOPS). Updated the sizing comment accordingly.

- Unsupported volume types: Standard persistent disks (`pd-standard`, HDD-backed) cannot meet Zeebe's Raft flush latency requirements. Use SSD-backed (`pd-ssd`) volumes.

> <!-- Sizing note: on GKE pd-ssd, IOPS and throughput both scale with capacity (~30 IOPS and ~0.48 MB/s per GiB; https://cloud.google.com/compute/docs/disks/performance). Reaching 3,000 IOPS needs ~100 GiB and 125 MiB/s needs ~260 GiB, so ~260 GiB (throughput-bound) sustains both baselines. -->

Reviewer: Fixed in 92ca16882 — removed the exact GiB figure and the MiB/MB derivation entirely. The GKE section now states the principle (on `pd-ssd`, IOPS and throughput scale with disk size; throughput is the binding constraint) and links to GCP disk performance, so there is no unit ambiguity.
